controllers/front/AuthController.php

```diff
@@ -41,49 +40,25 @@ public function checkAccess()
 
     public function initContent()
     {
-        $should_redirect = false;
-
-        if (Tools::isSubmit('submitCreate') || Tools::isSubmit('create_account')) {
-            $register_form = $this
-                ->makeCustomerForm()
-                ->setGuestAllowed(false)
-                ->fillWith(Tools::getAllValues());
-
-            if (Tools::isSubmit('submitCreate')) {
-                $hookResult = array_reduce(
-                    Hook::exec('actionSubmitAccountBefore', [], null, true),
-                    function ($carry, $item) {
-                        return $carry && $item;
-                    },
-                    true
-                );
-                if ($hookResult && $register_form->submit()) {
-                    $should_redirect = true;
-                }
-            }
+        if (Tools::isSubmit('create_account')) {
+            return $this->redirectWithNotifications('registration');
```

Comment:
@jolelievre This redirection works perfectly. We were talking about a situation when you do:
- `$this->context->link->getPageLink('authentication', true, null, ['create_account' => '1']);`
- Pass this URL to smarty for your login popup.
- Make a POST request to it.

We could fix it by returning a correct URL in getPageLink, WDYT?

Then the only way it could fail is, if somebody was POSTing to hardcoded `/login?create_account` for whatever reason. 😄

Reply:
Actually, the case with hardcoded URL is exactly what I had in mind In this case we need to handle the POST redirection So we are sure that this modification from your PR is fully backward compatible

And this way no need to handle a tricky edge case in `getPageLink`